MSAL/src/MSALAccount.m

```diff
@@ -76,25 +73,47 @@ - (id)initWithUsername:(NSString *)username
         _homeAccountId = [[MSALAccountId alloc] initWithAccountIdentifier:homeAccountId
                                                                  objectId:uid
                                                                  tenantId:utid];
-        
-        _localAccountId = [[MSALAccountId alloc] initWithAccountIdentifier:localAccountId
-                                                                  objectId:localAccountId
-                                                                  tenantId:tenantId];
 
         _lookupAccountIdentifier = [[MSIDAccountIdentifier alloc] initWithDisplayableId:username homeAccountId:homeAccountId];
+        
+        if (tenantProfiles.count > 0)
+        {
+            _tenantProfiles = [[NSMutableArray alloc] initWithArray:tenantProfiles];
+        }
     }
 
     return self;
 }
 
 - (id)initWithMSIDAccount:(MSIDAccount *)account
+      createTenantProfile:(BOOL)createTenantProfile
 {
+    NSError *error;
+    MSALAuthority *authority = [MSALAuthorityFactory authorityFromUrl:account.authority.url context:nil error:&error];
+    if (error || !authority)
+    {
+        MSID_LOG_ERROR(nil, @"Failed to create msal authority from msid authority!");
+        return nil;
+    }
+    
+    NSArray *tenantProfiles;
+    if (createTenantProfile)
+    {
+        MSALTenantProfile *tenantProfile = [[MSALTenantProfile alloc] initWithUserObjectId:account.localAccountId
+                                                                                  tenantId:account.tenantId
+                                                                                 authority:authority
+                                                                              isHomeTenant:account.isHomeTenantAccount
+                                                                                    claims:account.idTokenClaims.jsonDictionary];
```

Comment:
Will id token claims always be filled? I only saw us setting them in Legacy token cache accessor.

Reply:
addressed, now will fill id token claims for default access as well :)